### DISCUSSION TITLE
Feature Request

### DISCUSSION THREAD

**Discussion by JamesIgoe** [MicrosoftEdge/WebView2Feedback#3897] [Q&A]
I am working win WebView2 in a WPF application, and it is designed to accept configuration parameters via command line or file, a reusable utility for a locked-down environment.

The one complaint we are getting from doctors is that the MS login gets 'forgotten' after the browser window closes, requiring that they login via a popup MS window. If a Window that knows the user identity is open,  it seems to reuse that authentication.

Is there a way to persist that login, so that after the window closes, subsequent windows reuse that login, along with maybe an expiration on the file(s) that have that information?

**Comment by champnic** [comment]
@JamesIgoe 
Hey James,
Are you seeing a pattern about when the info is forgotten vs remembered? Session info and cookies are saved to the user data folder and should persist across sessions, unless you are deleting the user data folder or the cookies are set to expire.

You also mentioned "browser window" and I'm not sure if that's referring to your app's window or the actual Edge browser. If Edge, WebView2 doesn't share info or state with the browser.

You might also be able to use Single Sign-on for the logged in Windows user to streamline this:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2environmentoptions.allowsinglesignonusingosprimaryaccount?view=webview2-dotnet-1.0.902.49
